pkgs/servers/http/nginx/generic.nix

```diff
@@ -68,6 +68,14 @@ stdenv.mkDerivation {
     "--with-http_stub_status_module"
     "--with-threads"
     "--with-pcre-jit"
+    "--http-log-path=/var/log/nginx/access.log"
+    "--error-log-path=/var/log/nginx/error.log"
+    "--pid-path=/var/log/nginx/nginx.pid"
+    "--http-client-body-temp-path=/var/cache/nginx/client_body"
+    "--http-proxy-temp-path=/var/cache/nginx/proxy"
+    "--http-fastcgi-temp-path=/var/cache/nginx/fastcgi"
+    "--http-uwsgi-temp-path=/var/cache/nginx/uwsgi"
+    "--http-scgi-temp-path=/var/cache/nginx/scgi"
```

Comment:
Aren't these three directives more for IPC with other processes, and probably should be moved to RuntimeDirectory? WDYT, @aanderse ?

Reply:
If I had to guess I would agree what you said sounds most likely. I suggest someone confirm the purpose of those options.

Reply:
These folders use nginx only:
http://nginx.org/r/fastcgi_temp_path
http://nginx.org/r/scgi_temp_path
http://nginx.org/r/uwsgi_temp_path

Reply:
Thanks for clarification @Izorkin!
I assume a `cache` directory was selected over `/tmp` because the lack of `PrivateTmp`?

Reply:
No.
The `cache` directory should not depend on `PrivateTmp`.

Reply:
So they're not in `/run` or `/tmp`, as they can grow very large?

Reply:
Probably. I did not come across in practice.

Reply:
After restarting `nginx`, the `cache` directory in `/tmp` will be deleted.